### PR TITLE
YarnSize can be provided as Dtex in the API and displayed in both Nm and Dtex in the UI

### PR DESCRIPTION
**Pull request by magopian**
[user story](https://www.notion.so/fd0068a7db8348c584cd53a3eea399f2?v=37bf8aeb782747c3bf744ecaeafefd04&p=81e70f51d0f64ba799d6bd2fdae66bdd&pm=s)

### CODE REVIEW

**Comment by n1k0** [comment]
Rendering issue 
image

**Comment by magopian** [comment]
Indeed, thanks! Just fixed it 🙏

File: src/Server/Query.elm [hunk 1]
```diff
@@ -748,28 +749,69 @@ maybeMakingWasteParser key =
...
 
-maybeYarnSize : String -> Parser (ParseResult (Maybe Unit.YarnSize))
-maybeYarnSize key =
-    Query.int key
+parseYarnSize : String -> Maybe Unit.YarnSize
+parseYarnSize str =
+    let
+        withUnitRegex =
+            -- Match either an int or a int and a unit `Nm` or `Dtex`
+            Regex.fromString "(\\d+)\\s?(Nm|Dtex)"
```
Comment (n1k0) [inline src/Server/Query.elm:757]: I'd disallow the optional space between the number and the unit, especially as a query string parameter value

**Review by n1k0 — APPROVED** [review]
Great work!

File: src/Server/Query.elm (hunk 1 above)
Comment (magopian) [reply src/Server/Query.elm]: good point, thanks

**Pull request merged**